File: 150.Evaluate.Reverse.Polish.Notation.cpp
class Solution {
public:

int evalRPN(vector<string>& tokens) {
	stack<long>st;
	long l, r;
	for (auto t : tokens)
	{
		char firstchar = '0';
		if (t.length() == 1)
		{
			firstchar = t[0];
		}
		switch (firstchar)
		{
		case '+':
			r = st.top(); st.pop();
			l = st.top(); st.pop();
			st.push(l + r);
			break;
		case '-':

			r = st.top(); st.pop();
			l = st.top(); st.pop();
			st.push(l - r);
			break;
		case '*':
			r = st.top(); st.pop();
			l = st.top(); st.pop();
			st.push(l*r);
			break;
		case '/':
			r = st.top(); st.pop();
			l = st.top(); st.pop();
			st.push(l / r);
			break;
		default:
			l = stol(t);
			st.push(l);
			break;
		}
	}
	return st.top();
}

};

File: 150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        q = []
        for tok in tokens:
            #print(q)
            if tok == '+':
                q.append(q.pop() + q.pop())
            elif tok == '-':
                b, a = q.pop(), q.pop()
                q.append(a - b)
            elif tok == '*':
                q.append(q.pop() * q.pop())
            elif tok == '/':
                b, a = q.pop(), q.pop()
                q.append(int(a/b))
            else:
                q.append(int(tok))
        return q[0]